src/Aspire.Hosting.Garnet/GarnetBuilderExtensions.cs

```diff
@@ -128,20 +128,43 @@ public static IResourceBuilder<GarnetResource> WithDataBindMount(this IResourceB
     /// <code lang="csharp">
     /// var cache = builder.AddGarnet("cache")
     ///                    .WithDataVolume()
-    ///                    .WithPersistence(TimeSpan.FromSeconds(10), 5);
+    ///                    .WithPersistence(TimeSpan.FromSeconds(10));
     /// </code>
     /// </example>
     /// <param name="builder">The resource builder.</param>
     /// <param name="interval">The interval between snapshot exports. Defaults to 60 seconds.</param>
     /// <param name="keysChangedThreshold">The number of key change operations required to trigger a snapshot at the interval. Defaults to 1.</param>
     /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    [Obsolete("This method is obsolete and will be removed in a future version. Use the overload without the keysChangedThreshold parameter.")]
     public static IResourceBuilder<GarnetResource> WithPersistence(this IResourceBuilder<GarnetResource> builder,
-        TimeSpan? interval = null, long keysChangedThreshold = 1)
-        => builder.WithAnnotation(new CommandLineArgsCallbackAnnotation(context =>
-        {
-            context.Args.Add("--save");
-            context.Args.Add((interval ?? TimeSpan.FromSeconds(60)).TotalSeconds.ToString(CultureInfo.InvariantCulture));
-            context.Args.Add(keysChangedThreshold.ToString(CultureInfo.InvariantCulture));
-            return Task.CompletedTask;
-        }), ResourceAnnotationMutationBehavior.Replace);
+        TimeSpan? interval, long keysChangedThreshold)
+        => WithPersistence(builder, interval);
+
+    /// <summary>
+    /// Configures a Garnet container resource for persistence.
+    /// </summary>
+    /// <example>
+    /// Use with <see cref="WithDataBindMount(IResourceBuilder{GarnetResource}, string, bool)"/>
+    /// or <see cref="WithDataVolume(IResourceBuilder{GarnetResource}, string?, bool)"/> to persist Garnet data across sessions with custom persistence configuration, e.g.:
+    /// <code lang="csharp">
+    /// var cache = builder.AddGarnet("cache")
+    ///                    .WithDataVolume()
+    ///                    .WithPersistence(TimeSpan.FromSeconds(10));
+    /// </code>
+    /// </example>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="interval">The interval between snapshot exports. Defaults to 60 seconds.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<GarnetResource> WithPersistence(this IResourceBuilder<GarnetResource> builder,
+        TimeSpan? interval = null) =>
+    builder.WithAnnotation(new CommandLineArgsCallbackAnnotation(context =>
+    {
+        context.Args.Add("--checkpointdir");
+        context.Args.Add("/data/checkpoints");
```

Comment:
Just curious - what is this path?

Reply:
Its the path that garnet save aof log files.

Reply:
Is it something that we are choosing arbitrarily or is that a known path for garnet?

Reply:
`/data` is khow path in Garnet dockerfile since https://github.com/microsoft/garnet/pull/539. Garnet container run as none root user.

Reply:
Why are we overriding this though? The [documentation](https://microsoft.github.io/garnet/docs/getting-started/configuration) says `Storage directory for checkpoints. Uses logdir if unspecified.`. Is that not good enough?

I don't really know anything about this, so asking questions 😬

Reply:
I removed these lines and the container failed to start.
```csharp
        context.Args.Add("--checkpointdir");
        context.Args.Add("/data/checkpoints");
```
Garnet container logs:

```

/_||___||_\ Garnet 1.0.17 64 bit; standalone mode

'. \ / .' Port: 6379

'.\ /.' https://aka.ms/GetGarnet

'.'



Unable to initialize server due to exception: Access to the path '/app/AOF' is denied.
```
Garnet container runs as none root user, so in https://github.com/microsoft/garnet/pull/539, they give permission to this user only for the `/data` directory.

The [documentation](https://microsoft.github.io/garnet/docs/getting-started/configuration) should be updated.